disease/etl/omim.py

```diff
@@ -0,0 +1,106 @@
+"""Module to load disease data from OMIM."""
+import logging
+from .base import Base
+from disease import PROJECT_ROOT, DownloadException
+from disease.schemas import Meta, SourceName, Disease, NamespacePrefix
+from disease.database import Database
+from pathlib import Path
+from typing import List
+
+logger = logging.getLogger('disease')
+logger.setLevel(logging.DEBUG)
+
+
+class OMIM(Base):
+    """Gather and load data from OMIM.
+
+    MIM number prefix:
+    ------------------
+    Asterisk (*)  Gene (exclude)
+    Plus (+)  Gene and phenotype, combined  (exclude)
+    Number Sign (#)  Phenotype, molecular basis known
+    Percent (%)  Phenotype or locus, molecular basis unknown
+    NULL (<null>)  Other, mainly phenotypes with suspected mendelian basis
+    Caret (^)  Entry has been removed from the database or moved to another
+    entry (exclude)
+    """
+
+    def __init__(self,
+                 database: Database,
+                 src_url: str = "https://www.omim.org/downloads",
+                 data_path: Path = PROJECT_ROOT / 'data' / 'omim'):
+        """Override base class init method.
+
+        :param therapy.database.Database database: app database instance
+        :param pathlib.Path data_path: path to local OMIM data directory
+        """
+        self._SRC_URL = src_url
+        super().__init__(database=database, data_path=data_path)
+
+    def perform_etl(self) -> List[str]:
+        """Public-facing method to initiate ETL procedures on given data.
+
+        :return: empty list (because OMIM IDs shouldn't be used to construct
+            merged concept groups)
+        """
+        try:
+            self._extract_data()
+        except DownloadException:
+            logger.error("OMIM data extraction failed: input file must be "
+                         "manually placed in data directory.")
+        self._load_meta()
+        self._transform_data()
+        self.database.flush_batch()
+        return []
+
+    def _download_data(self):
+        """Download OMIM source data for loading into normalizer."""
+        raise DownloadException("OMIM data not available for public download")
```

Comment:
The `DownloadException` in `_download_data` does not display. You should add in the readme about users having to request OMIM data, which file to use, where to move the file and what to name it in the readme. 
```suggestion
        try:
            self._extract_data()
        except DownloadException:
            logger.error("OMIM data extraction failed: input file must be "
                         "manually placed in data directory.")
            raise DownloadException("OMIM data not available for public download")
        self._load_meta()
        self._transform_data()
        self.database.flush_batch()
        return []

    def _download_data(self):
        """Download OMIM source data for loading into normalizer."""
        raise DownloadException()
```